src/miral/display_configuration.cpp

```diff
@@ -20,77 +20,59 @@
 #include "static_display_config.h"
 
 #include <mir/server.h>
-#include <mir/log.h>
 
-#include <unistd.h>
+#include <boost/throw_exception.hpp>
 
 #include <fstream>
 #include <sstream>
 
-struct miral::DisplayConfiguration::Self : StaticDisplayConfig
+using namespace std::string_literals;
+
+class miral::DisplayConfiguration::Self : public ReloadingYamlFileDisplayConfig
 {
-    Self(std::string const& name) : name{name} {}
-    std::string const name;
+public:
 
-    void dump_config(std::function<void(std::ostream&)> const& print_template_config) override
+    Self(std::string const& basename) : ReloadingYamlFileDisplayConfig{basename}
     {
-        std::string config_dir;
+        std::string config_roots;
 
         if (auto config_home = getenv("XDG_CONFIG_HOME"))
-            config_dir = config_home;
+        {
+            (config_roots = config_home) += ":";
+            config_path(config_home);
+        }
         else if (auto home = getenv("HOME"))
-            (config_dir = home) += "/.config";
-
-        if (config_dir.empty())
         {
-            mir::log_debug("Nowhere to write display configuration template: Neither XDG_CONFIG_HOME or HOME is set");
+            (config_roots = home) += "/.config:";
+            config_path(home + "/.config"s);
         }
+
+        if (auto config_dirs = getenv("XDG_CONFIG_DIRS"))
+            config_roots += config_dirs;
         else
+            config_roots += "/etc/xdg";
```

Comment:
> There's a bit of a confusing behaviour with `miral::DisplayConfiguration` in that it will only look at the different roots once, and not load the configuration from a location higher in the lookup list if it appeared there. Something to be documented, at least - but we may need to deal with it in Frame when ran as a user.

Good suggestion. At present we just watch the file we found.

Watching locations where we didn't find a file is possible, but then do we also want to watch for those locations appearing? E.g. if `~/.config' doesn't exist do we first monitor `~` for the creation of `.config` and then monitor that?

Reply:
> Watching locations where we didn't find a file is possible,

I'm not _sure_ we want it… IMO documenting that behaviour would be enough.

> do we also want to watch for those locations appearing

But if you think worth it - I don't think we need to go that far… If a root doesn't exist, log/warn about it and skip it.